eisbuk-admin/src/components/atoms/IntervalCard/__tests__/IntervalCard.test.tsx

```diff
@@ -1,16 +1,75 @@
 import React from "react";
-import { render } from "@testing-library/react";
+import {cleanup, screen render } from "@testing-library/react";
 
 import IntervalCard from "../IntervalCard";
+import * as bookingActions from "@/store/actions/bookingOperations";
+import { bookedSlot, nonBookedSlot } from "../__testData__/dummyData";
 
-import { baseProps } from "../__testData__/dummyData";
+import { Customer, Slot } from "eisbuk-shared";
+import { SlotInterface, SlotInterval } from "@/types/temp";
+
+
+const mockDispatch = jest.fn();
+
+jest.mock("react-redux", () => ({
+  ...jest.requireActual("react-redux"),
+  useDispatch: () => mockDispatch,
+  /** We're mocking this not to fail tests including SlotForm (this will be returning newSlotTime as null) @TODO remove in the future */
+  useSelector: () => null,
+}));
+
+jest.mock("react-router-dom", () => ({
+  ...jest.requireActual("react-router-dom"),
+  useParams: () => ({ secretKey: "secret_key" }),
+}));
+
+/** @TODO remove this when the i18next is instantiated with tests */
+jest.mock("i18next", () => ({
+  ...jest.requireActual("i18next"),
+  /** We're mocking this not to fail certain tests depending on this, but we're not testing the i18n values, so this is ok for now @TODO fix i18n in tests */
+  t: () => "",
+}));
```

Comment:
You will actually need this. I would suggest to mock `t` implementation as `(label: string) => label`. This way, in test DOM, the appropriate value for i18n label will be rendered and you can use `screen.getByText(__bookInterval__)` for instance and have `const __bookInterval__ = "BookingCard.BookInterval"` saved in `@/lib/labels.ts` and in translation files have entries
```json
// en.json
"BookingCard": {
  "BookInterval": "BOOK INTERVAL"
}

// it.json
"BookingCard": {
  "BookInterval": "PRENOTA INTERVAL"
}
```
This way the testing is straightforward and you're reducing chances for typos.